nautobot/extras/tests/git_helper.py

```diff
@@ -69,6 +69,12 @@ def create_and_populate_git_repository(target_path):
         # Directory "01-valid-files" --> tag "valid-files" so that we won't break tests if we renumber the directories
         repo.create_tag(dirname[3:], message=f"Tag based on {dirname} files")
 
+    if divergent_branch:
+        repo.create_head(divergent_branch)
+        repo.head.reference = repo.heads[divergent_branch]
+        repo.head.reset(index=True, working_tree=True)
+        repo.create_tag(f"{divergent_branch}", message=f"Tag for divergent branch {divergent_branch}")
```

Comment:
Not sure I entirely understand what this block is actually doing, but is the resulting divergent branch any way different in content from the main branch (i.e., at least a different commit hash)?